tests/storage-compat/storage-compatibility.sh

```diff
@@ -17,48 +17,52 @@ git lfs pull
 # Uncompress snapshot storage
 tar -xvjf ./tests/storage-compat/storage.tar.bz2
 
-# Run in background
-./target/debug/qdrant &
+# Test it boots up fine with the old storage
+./target/debug/qdrant & 
 
-# Sleep to make sure the process has started (workaround for empty pidof)
-sleep 5
-
-## Capture PID of the run
-PID=$(pidof "./target/debug/qdrant")
-echo $PID
+sleep 1
 
+declare retry=0
 until curl --output /dev/null --silent --get --fail http://$QDRANT_HOST/collections; do
-  printf 'waiting for server to start...'
-  sleep 5
+  if ((retry++ < 30)); then
+      printf 'waiting for server to start...'
+      sleep 1
+  else
+      echo "Collection failed to load in ~30 seconds" >&2
+      exit 2
+  fi
 done
 
 echo "server ready to serve traffic"
 
 echo "server is going down"
-kill -9 $PID
+pidof qdrant | xargs kill -9
 echo "END"
 
 
 # Test recovering from an old snapshot
-gzip -d --keep ./tests/storage-compat/collection.snapshot.gz
+gzip -d --keep ./tests/storage-compat/full-snapshot.snapshot.gz
 
 rm -rf ./storage
-./target/debug/qdrant --snapshot ./tests/storage-compat/collection.snapshot:test_collection &
-
-# Sleep to make sure the process has started (workaround for empty pidof)
-sleep 5
-
-## Capture PID of the run
-PID=$(pidof "./target/debug/qdrant")
-echo $PID
+./target/debug/qdrant \
+  --storage-snapshot ./tests/storage-compat/full-snapshot.snapshot &
 
+declare retry=0
 until curl --output /dev/null --silent --get --fail http://$QDRANT_HOST/collections/test_collection; do
-  printf 'waiting for server to start...'
-  sleep 5
+  if ((retry++ < 30)); then
+      printf 'waiting for server to start...'
+      sleep 1
+  else
+      echo "Collection failed to load in ~30 seconds" >&2
+      exit 2
+  fi
 done
 
 echo "server ready to serve traffic"
 
 echo "server is going down"
-kill -9 $PID
+pidof qdrant | xargs kill -9
```

Comment:
this might kill unrelated instances

Reply:
Must be because I'm on mac, but `pidof "./target/debug/qdrant"` does not bring anything for me. I'll look for a better solution

Reply:
@generall finally resorted to capturing the PID via `$!`